doc/devel/contribute.rst

```diff
@@ -367,26 +367,154 @@ project that leads to a scientific publication, please follow the
 Coding guidelines
 =================
 
-API changes
------------
+While the current state of the Matplotlib code base is not compliant with all
+of these guidelines, but our goal in enforcing these constraints on new
+contributions is that it improves the readability and consistency of the code base
+going forward.
+
+PEP8, as enforced by flake8
+---------------------------
+
+Formatting should follow the recommendations of PEP8_, as enforced by flake8_.
+Matplotlib modifies PEP8 to extend the maximum line length to 88
+characters. You can check flake8 compliance from the command line with ::
+
+    python -m pip install flake8
+    flake8 /path/to/module.py
+
+or your editor may provide integration with it.  Note that Matplotlib intentionally
+does not use the black_ auto-formatter (1__), in particular due to its inability
+to understand the semantics of mathematical expressions (2__, 3__).
+
+.. _PEP8: https://www.python.org/dev/peps/pep-0008/
+.. _flake8: https://flake8.pycqa.org/
+.. _black: https://black.readthedocs.io/
+.. __: https://github.com/matplotlib/matplotlib/issues/18796
+.. __: https://github.com/psf/black/issues/148
+.. __: https://github.com/psf/black/issues/1984
+
+
+Package imports
+---------------
+Import the following modules using the standard scipy conventions::
+
+  import numpy as np
+  import numpy.ma as ma
+  import matplotlib as mpl
+  import matplotlib.pyplot as plt
+  import matplotlib.cbook as cbook
+  import matplotlib.patches as mpatches
+
+In general, Matplotlib modules should **not** import `.rcParams` using ``from
+matplotlib import rcParams``, but rather access it as ``mpl.rcParams``.  This
+is because some modules are imported very early, before the `.rcParams`
+singleton is constructed.
+
+Variable names
+--------------
+
+When feasible, please use our internal variable naming convention for objects
+of a given class and objects of any child class:
+
++---------------+------------------------------------+
+| variable name |             base class             |
++===============+====================================+
+| ``fig``       | `~matplotlib.figure.FigureBase`    |
++---------------+------------------------------------+
+| ``ax``        | `~matplotlib.axes.Axes`            |
++---------------+------------------------------------+
+| ``trans``     | `~matplotlib.transforms.Transform` |
++---------------+------------------------------------+
+
+Generally, denote more than instance of the same class by adding numbers to
+the variable names, i.e. ``trans1``, ``trans2``.
```

Comment:
Can we please suggest naming that is meaningful rather than just numbers?

Reply:
anything but numbers!  `trans_<from>_to_<target>` ? All of the `trans<name>` attributes are implicitly to screen so (I assume) we dropped the target.  In function bodies where transformations may be targeting other coordinate systems I think we should include both.